test/controllers/api-key.test.js

```diff
@@ -24,60 +34,167 @@ describe('ApiKeyController tests', () => {
+      expect(response.status).to.equal(STATUS_BAD_REQUEST);
+    });
+
+    it('should throw an error if request data is missing features', async () => {
+      requestContext.data = { domains: ['https://example.com'] };
       const response = await apiKeyController.createApiKey({ ...requestContext });
-      expect(response.status).to.equal(501);
+      expect(response.status).to.equal(STATUS_BAD_REQUEST);
     });
 
-    it('should throw a not implemented error when data is null', async () => {
-      requestContext.data = null;
+    it('should throw an error if request data is missing domains', async () => {
+      requestContext.data = { features: ['imports'] };
       const response = await apiKeyController.createApiKey({ ...requestContext });
-      expect(response.status).to.equal(501);
+      expect(response.status).to.equal(STATUS_BAD_REQUEST);
+    });
+
+    it('should throw an error if request data is missing name', async () => {
+      requestContext.data = { features: ['imports'], domains: ['https://example.com'] };
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_BAD_REQUEST);
+    });
+
+    it('should throw an error if the domain is not valid', async () => {
+      requestContext.data.domains = ['invalid-domain'];
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_BAD_REQUEST);
+    });
+
+    it('should throw an error if imsOrgId is missing', async () => {
+      requestContext.pathInfo.headers['x-ims-gw-org-id'] = '';
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_UNAUTHORIZED);
+    });
+
+    it('should throw an error if the organization is not found', async () => {
+      context.imsClient.getImsUserProfile.returns({ organizations: [] });
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_UNAUTHORIZED);
+    });
+
+    it('should throw an error if bearer token is missing', async () => {
+      requestContext.pathInfo.headers.authorization = '';
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_UNAUTHORIZED);
+    });
+
+    it('should create a new API key', async () => {
+      context.dataAccess.getApiKeysByImsUserIdAndImsOrgId.returns([]);
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_CREATED);
+    });
+
+    it('should throw an error if the number of domains exceeds the limit', async () => {
+      requestContext.data.domains = ['https://example.com', 'https://another.com'];
+      const response = await apiKeyController.createApiKey({ ...requestContext });
+      expect(response.status).to.equal(STATUS_FORBIDDEN);
```

Comment:
Can we check the `x-error` header values here, too?

Reply:
Done.